prereise/gather/hiflddata/data_trans.py

```diff
@@ -0,0 +1,433 @@
+            continue
+        neigh = get_neigbors(G, sub, depth=5)
+        temp_KV = 0
+        num = 0
+        for i in range(1, 6):
+            for nei in neigh[i]:
+                if (nei in KV_dict):
+                    temp_KV = temp_KV + KV_dict[nei]
+                    num = num + 1
+            if (num > 0):
+                KV_dict[sub] = temp_KV / num
+                continue
+            else:
+                KV_dict[sub] = -999999
+    return KV_dict
+
+
+def Set_Sub(E_csv):
+    sub_by_coord_dict = {}
+    sub_name_dict = {"sub": []}
+    raw_subs = pd.read_csv(E_csv)
+    for index, row in raw_subs.iterrows():
+        location = (row['LATITUDE'], row['LONGITUDE'])
+        if location in sub_by_coord_dict:
+            print("WARNING: substations coordinates conflict check:", location)
+        sub_by_coord_dict[location] = (row['ID'], row['NAME'])
+        if row['NAME'] not in sub_name_dict:
+            sub_name_dict[row['NAME']] = []
+        sub_name_dict[row['NAME']].append((row['LATITUDE'], row['LONGITUDE']))
+    return sub_by_coord_dict, sub_name_dict
+
+# Write sub.csv
+def Write_sub(clean_data,zone_dic):
+    sub = open('output/sub.csv','w',newline='')
+    csv_writer = csv.writer(sub)
+    csv_writer.writerow(["sub_id","sub_name","lat","lon","zone_id","type"])
+    for index, row in clean_data.iterrows():
+        csv_writer.writerow([row['ID'],row['NAME'],row['LATITUDE'],row['LONGITUDE'],zone_dic[row['STATE']],row['TYPE']])
+    sub.close()
+
+
+#Write bus.csv
+def Write_Bus(clean_data, zone_dic, KV_dict):
+    bus = open('output/bus.csv', 'w', newline='')
+    csv_writer = csv.writer(bus)
+    csv_writer.writerow(["Bus_id", "PD", "Zone_id", "base_KV"])
+    missingSub = []
+    for index, row in clean_data.iterrows():
+        sub = (row['LATITUDE'], row['LONGITUDE'])
+        if (sub in KV_dict):
+            csv_writer.writerow([row['ID'], 0, zone_dic[row['STATE']], KV_dict[sub]])
+        else:
+            missingSub.append(sub)
+    bus.close()
+    print("INFO: ", len(missingSub), " substations excluded from the network. Some examples:")
+    print(missingSub[:20])
+
+
+#Write bus2sub.csv
+def Write_bus2sub(clean_data):
+    bus2sub = open('output/bus2sub.csv','w',newline='')
+    csv_writer = csv.writer(bus2sub)
+    csv_writer.writerow(["Bus_id","sub_id"])
+    for index, row in clean_data.iterrows():
+        csv_writer.writerow([row['ID'],row['ID']])
+    bus2sub.close()
+
+
+#Write branch.csv
+def Write_branch(lines):
+    branch = open('output/branch.csv', 'w', newline='')
+    csv_writer = csv.writer(branch)
+    csv_writer.writerow(['branch_id', 'line_type', 'from_bus_id', 'from_bus_name', 'to_bus_id', 'to_bus_name', 'length_in_mile', 'reactance', 'rateA'])
+    csv_writer.writerows(lines)
+    branch.close()
+
+# Reactance Calculate 1
+kv_xperunit_calculate_1 = (
+    (69, 0.2293),
+    (115, 0.1106),
+    (230, 0.0567),
+    (345, 0.0074),
+    (500, 0.0001),
+    (2147483647, 0.0001)
```

Comment:
This is a very high voltage!

Reply:
Thanks. This was only to represent the case that VOLTAGE is >500 as the boundary condition. The highest VOLTAGE in HIFLD transmission line data is 1000. I am changing this to 1100 to avoid potential confusions.

Reply:
It would be nice if we can avoid mixing fake data in with real data, and algorithmically handle the case where a line voltage is higher than what we have in the data.